src/lib/syncable_ledger/syncable_ledger.ml

```diff
@@ -23,6 +23,12 @@ module Query = struct
         | Num_accounts
             (** How many accounts are there? Used to size data structure and
             figure out what part of the tree is filled in. *)
+        | Subtree of 'addr
+            (** What are the 2^k nodes at depth k from the given prefix 
+            address **)
+      (* TODO: Properly handle versioning *)
+      (* TODO: Consider additional query to verify subtree suport, for
+         softfork compatibility *)
```

Comment:
Unfortunately, we aren't able to modify this type here, we'll have to add a converter. Some nodes won't have upgraded, so we need to be able to have a fallback.

For example, it might be natural to change e.g.
```ocaml
| What_child_hashes of 'addr
```
into e.g.
```ocaml
| What_child_hashes of {address: 'addr; depth: int}
```
and then the response can be tweaked from
```ocaml
| Child_hashes_are of 'hash * 'hash
```
to
```ocaml
| Child_hashes_are of 'hash array
```

I think this gives a natural downgrade path
```ocaml
function
| What_child_hashes {address; depth = _} -> V1.What_child_hashes address
| ...
```
and upgrade for the response of
```ocaml
function
| Child_hashes_are (x, y) -> V2.Child_hashes_are [x; y]
| ...
```

As currently written, a node running this code won't be able to sync to the network :'(

Reply:
To be sure I was thinking the same, for now I added it in V1 just to make it compile until I understand better how versioning works, but I agree it should fail right now.
But if I add a V2 I can add another variant or make arbitrary changes to the type, right? or are there additional limitations when making a new version? and for that reason you suggest modifying `Child_hashes_are` instead?

Reply:
I think I understand the example, but I see a possible issue:
Send V2 request, receiver converts it into V1 and sends back V1 response, that is converted to V2 response after arrival, although a trivial one.
But in the opposite case sending a V1 and receiving a V2 response shouldn't work, as the V1 node can convert V2 back to V1.
Or does the example assume both types have the same representation and a V2 can be deserialized as V1 without knowing anything about V2?

Reply:
The versioning system should ensure that a `V1` request never receives a `V2` response. The RPC protocol does version negotiation.

Reply:
For reference, [this](https://github.com/MinaProtocol/mina/blob/2c77413cb6e93361d88d2d70acf2aa23aa67843a/src/lib/mina_networking/rpcs.ml#L369) is where the version negotiation happens.